### DISCUSSION TITLE
Future of Fooocus-API

### DISCUSSION THREAD

**Discussion by davefojtik** [mrhan1993/FooocusWithFastAPI#34] [General]
Hello @mrhan1993. Since there are now two repos: [FooocusAPI](https://github.com/mrhan1993/FooocusAPI/) and [Fooocus-API](https://github.com/mrhan1993/Fooocus-API) - (good job with the refactoring!), I would want to ask what are the plans for each of them. Particularly:
- Will both repos get updates in the future? Do you plan to discontinue the original one at some point?
- Would you recommend that everyone switch to the new refactored one?
- In the [documentation](https://github.com/mrhan1993/FooocusAPI/blob/main/docs/migration.md) of the refractored version you state that `require_base64` "will be removed". Does it mean we'll be unable to retrieve the image as base64, or rather be "default and non-selectable"?
- Are you in touch with maintainers about the possibility of merging the API into the core of the [Fooocus](https://github.com/lllyasviel/Fooocus)?

Thank you for your time on these projects and for answering our questions. I appreciate your hard work!

**Comment by mrhan1993** [comment]
I am doing a compatibility merger, merging [FooocusAPI](https://github.com/mran1993/FooocusAPI/) into [Fooocus-API](https://github.com/mran1993/Fooocus-API). After completion, the core code and logic of the two projects will be the same. Due to the advantages brought by refactoring, this work will be completed quickly.

Both projects should continue to be updated, because in most cases, only git pull upstream is needed for refactored updates.

For require_base64, maybe I shouldn't delete it. I'll add it back later, sorry

The last question is that there should be no merger, because I may not be guaranteed to have enough energy to follow the main line.

**Reply by davefojtik** [reply]
Hello again. I want to revisit this issue after the recent events. The main Fooocus project is currently without a maintainer and won't receive any updates, including Flux. If abadonned, will your API end with it? Or will you try to switch to some fork of Fooocus that still gets updates and even already supports Flux like [metercai/SimpleSDXL](https://github.com/metercai/SimpleSDXL)?

**Reply by mrhan1993** [reply]
Oh! I just learned about this. You came up with a great idea, I'll go and take a look this [metercai/SimpleSDXL](https://github.com/metercai/SimpleSDXL).However, due to work reasons, I cannot guarantee that I will do this。

**Reply by davefojtik** [reply]
I totally understand that. It just might be a pity to abaddon solutions we spent months on, especially after you refractored it so well for future updates, just to see [end of feature updates](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#project-status-limited-long-term-support-lts-with-bug-fixes-only) moments after. But as I said to RunPod users already, 

> This is the harsh reality of the AI world. Everything improves so fast, but that also means the software implementing it is ageing at the same speed.

No matter how you decide, it's been an honour to implement your API to RunPod, and I want to thank you again for your hard work. I wish you the best of luck in whatever will come next!

**Reply by mrhan1993** [reply]
Do you think encapsulating ComfyUI is an acceptable option?ComfyUI can implement almost all the features of Fooocus through workflow, although I am still researching this.

**Reply by davefojtik** [reply]
I've been considering switching to ComfyUI too. Although it would probably make our repos obsolete anyway, as ComfyUI has both its [API](https://github.com/comfyanonymous/ComfyUI/blob/master/script_examples/basic_api_example.py) and [RunPod workers](https://github.com/blib-la/runpod-worker-comfy) available. Or do you have something better than the built-in API in mind?
